### PR TITLE
splitted parser into two versions

### PR DESCRIPTION
**Pull request by GuyAv46**
`RSQuery_ParseRaw` is now `RSQuery_ParseRaw_v1` or `RSQuery_ParseRaw_v2`,
All the calls are now to `RSQuery_ParseRaw_v2` which is the 2.4.3 parser with a hard coded flag

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -85,7 +85,8 @@ file(GLOB RS_SRC
     "src/aggregate/reducers/*.c"
     "src/ext/*.c"
     "src/hll/*.c"
-    "src/query_parser/*.c"
+    "src/query_parser/parser2.2/*.c"
+    "src/query_parser/parser2.4/*.c"
```
Comment (rafie) [inline CMakeLists.txt:89]: parser2.2 -> v1
perser2.4 -> v2
with matching to dialect.

File: tests/cpptests/test_cpp_query.cpp [hunk 2]
```diff
@@ -454,6 +721,61 @@ TEST_F(QueryTest, testFieldSpec) {
...
+  const char *qt = "@title:hello world";
+  QASTCXX ast(ctx);
+  int ver = 2;
+
+  ASSERT_TRUE(ast.parse(qt, ver)) << ast.getError();
+  //ast.print();
+  QueryNode *n = ast.root;
+  ASSERT_EQ(n->type, QN_PHRASE);
+  ASSERT_EQ(n->opts.fieldMask, -1);
+  ASSERT_EQ(n->children[0]->opts.fieldMask, 0x01);
```
Comment (DvirDukhan) [inline tests/cpptests/test_cpp_query.cpp:739]: can you verify the number of children here? and types?

**Review by DvirDukhan — APPROVED** [review]
looks good
small comment

File: tests/cpptests/test_cpp_query.cpp (hunk 2 above)
Comment (GuyAv46) [reply tests/cpptests/test_cpp_query.cpp:743]: improved. type of text leaves is always `QN_TOKEN` with or without modifier

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**